client/src/pages/NrMap.js

```diff
@@ -33,61 +33,75 @@ const Cards = styled.div`
   margin: 10px;
 `;
 
-class NrMap extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      latitude: 52.520008,
-      longitude: 13.404954,
-      rec: [],
-      plcDtl: [],
-    };
-    this.googleMapRef = React.createRef();
-  }
-  componentDidMount() {
+const map = {
+  googleMap: null,
+  marker: null,
+  place: null,
+};
+
+const NrMap = () => {
+  const [coordinates, setCoordinates] = useState({
+    latitude: 52.520008,
+    longitude: 13.404954,
+  });
+  // rec-> list of hospitals according to nearbySearch
+  const [rec, setRec] = useState([]);
+  // plcDtl-> list of hospitals with requested details
+  const [plcDtl, setPlcDtl] = useState(() => []);
+
+  const googleMapRef = useRef();
+
+  useEffect(() => {
     const googleMapScript = document.createElement("script");
     googleMapScript.src = url;
     window.document.body.appendChild(googleMapScript);
 
     googleMapScript.addEventListener("load", () => {
-      this.googleMap = this.createGoogleMap();
-      this.marker = this.createMarker();
-      this.place = this.places();
+      map.googleMap = createGoogleMap();
+      map.marker = createMarker();
+      map.place = places();
     });
-    this.getMyLocation();
-  }
-  createGoogleMap = () =>
-    new window.google.maps.Map(this.googleMapRef.current, {
+
+    getMyLocation();
+  }, []);
+
+  const createGoogleMap = () => {
+    return new window.google.maps.Map(googleMapRef.current, {
       zoom: 13,
       center: {
-        lat: this.state.latitude,
-        lng: this.state.longitude,
+        lat: coordinates.latitude,
+        lng: coordinates.longitude,
       },
       disableDefaultUI: true,
     });
-  places = () => {
-    new window.google.maps.places.PlacesService(this.googleMap).nearbySearch(
+  };
+
+  const places = () => {
+    new window.google.maps.places.PlacesService(map.googleMap).nearbySearch(
       {
         location: {
-          lat: this.state.latitude,
-          lng: this.state.longitude,
+          lat: coordinates.latitude,
+          lng: coordinates.longitude,
         },
         radius: 3000,
         type: ["hospital"],
       },
       (results, status) => {
         if (status !== "OK") return;
-        this.setState({ rec: results }, () => {
-          console.log(this.state.rec.place_id);
-          this.createMarker();
-          this.placeDetails();
-        });
+        setRec(results);
       },
     );
   };
 
-  placeDetails = () => {
-    this.state.rec.map((place) => {
+  useEffect(() => {
+    if (rec && rec.length > 0) {
+      placeDetails();
```

Comment:
Does the order matter for which function is called first? In the old changes, it creates a marker before getting place details.

Reply:
it doesn't matter. They are independent, one updates the google map and the other the component state.